### ISSUE TITLE
Disappeared draft.json file

### ISSUE THREAD

**Issue by JoeGermuska** [NUKnightLab/StoryMapJS#249]
A [Zendesk user reported](https://knightlab.zendesk.com/agent/tickets/1485) this error:

> Hi, I was nearly done with my map on refugees. Was attempting to use a GigaPixel one for a change. My initial map (called 'Refugees flows') was saved. So, when I attempted to resume editing, I got this error message.
> 
> May you please get back to me with an export of the file? I have lost a whole afternoon of work and have to resume everything from scratch!

Checking the records for this person's user ID shows that there are only HTML files on the S3 bucket, no JSON files.

Can you look for reasons why this might have happened, and for ways we can know when it happens, and of course, preferably prevent it?

```
appCodeName: Mozilla
appName: Netscape
appVersion: 5.0 (X11)
platform: Linux x86_64
userAgent: Mozilla/5.0 (X11; Linux x86_64; rv:38.0) Gecko/20100101 Firefox/38.0 Iceweasel/38.0.1
```

**Comment by jywsn** [comment]
Deleting a StoryMap only happens via the "Delete" menu item in the cog menus on the list of StoryMaps (which requires user confirmation).  If something is inadvertently getting deleted via some other means, I have yet to discover where that is happening.

**Comment by JoeGermuska** [comment]
Another [zendesk report](https://knightlab.zendesk.com/agent/tickets/1495)
> I created a Storymap seconds ago but can`t open it again. There is an error-code.

```
appCodeName: Mozilla 
appName: Netscape 
appVersion: 5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/43.0.2357.130 Safari/537.36 
platform: Win32 
userAgent: Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/43.0.2357.130 Safari/537.36
```

4eebb62c4814eda965ca25bfd9209922/multimediale-schweiz-1

Let's figure out some places to add logging/diagnostics to try to trap this.

**Comment by JoeGermuska** [comment]
Another [zendesk report](https://knightlab.zendesk.com/agent/tickets/1564)

> I was working on my project and now an error message is coming up saying that it doesn't exist! I can no longer access my project. Please help!

```
appCodeName: Mozilla 
appName: Netscape 
appVersion: 5.0 (iPad; CPU OS 8_4 like Mac OS X) AppleWebKit/600.1.4 (KHTML, like Gecko) CriOS/44.0.2403.65 Mobile/12H143 Safari/600.1.4 (000129) 
platform: iPad 
userAgent: Mozilla/5.0 (iPad; CPU OS 8_4 like Mac OS X) AppleWebKit/600.1.4 (KHTML, like Gecko) CriOS/44.0.2403.65 Mobile/12H143 Safari/600.1.4 (000129)
```
de8c39898290d5180ac714db00b25f99/tituba-battle-for-the-body

**Comment by scott2b** [comment]
Here is another: https://knightlab.zendesk.com/agent/tickets/3633

I am really unsure of how we can track down the cause of this. For this one, it appears that I will be able to recover the files from glacier, although I am still working on it. It is a long and manually intensive process.

The fact that this missing map is stored in our backups is an indicator that the problem is not a lack of being saved to begin with. Maps are in fact being deleted at some point. I know of no reason this would happen that is not related to the delete mechanism in the editor. I am considering the possibility of disabling delete altogether until we get a better handle on this. Thoughts?

**Comment by scott2b** [comment]
I am unaware of ongoing incidents of maps being simply deleted or data merely disappearing. What we have continued to see is the degradation of the json data file during save mishaps that were replacing the data with an HTML error page (see #322). This issue was fixed by 1a85abe048037c82749ebd23e94da1169b0bb9a6, so I'm calling this fixed for now.
